avc/annexb.go

```diff
@@ -95,6 +85,56 @@ func ConvertByteStreamToNaluSample(stream []byte) []byte {
 	return out
 }
 
+func getStartCodePositions(stream []byte) (scNalus []scNalu, minStartCodeLength int) {
+	streamLen := len(stream)
+	uintSize := int(unsafe.Sizeof(uint(0)))
+	streamLenLim := streamLen - (streamLen % uintSize) - uintSize
+	minStartCodeLength = 4
+
+	i := 0
+	for ; i < streamLenLim; i += uintSize {
+		x := *(*uint)(unsafe.Pointer(&stream[i]))
+		if ((x - 0x0101010101010101) & (^x) & 0x8080808080808080) != 0 {
```

Comment:
This most probably only works for 64-bit machines since it is an explicit 64-bit value.
We need some other code for 32-bit architectures.

It also needs some documentation since it is hard to understand.

Reply:
Good day, thanks for review!

I think we can declare this magic numbers as typed uint64 const's and then convert it to uint inside the if statement. As we didn't initialise uint to overflow value, but instead convert it from known type this should help. This purely my assumption.
I will fix all other issues and come back soon.

Reply:
Golang automatically converts constants including literals to what int type is needed, so a there is no way we can convert a 64-bit value with the any of the upper 32 bits set into into a 32-bit one. We could have a different 32-bit constant, but we should avoid a test inside the for loop. That is the main reason I'd rather make a separate function/flow for 32-bit platforms.

Reply:
Here is a reference to the magic line:
https://graphics.stanford.edu/~seander/bithacks.html#ZeroInWord

Reply:
I added comments to code and moved zero byte checking to separate function. Please review